platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -649,38 +698,181 @@ class HangingProtocolService {
     }
   }
 
-  private _setProtocol(
-    protocol: HangingProtocol.Protocol,
+  /**
+   * Updates the stage activation, setting the stageActivation values to
+   * 'disabled', 'active', 'passive' where:
+   * * disabled means there are insufficient viewports filled to show this
+   * * passive means there aren't enough preferred viewports filled to show
+   * this stage by default, but it can be manually selected
+   * * enabled means there are enough viewports to select this viewport by default
+   *
+   * The logic is currently simple, just count how many viewports would be
+   * filled, and compare to the required/preferred count, but the intent is
+   * to allow more complex rules in the future as required.
+   *
+   * @returns the stage number to apply initially, given the options.
+   */
+  private _updateStageActivation(
```

Comment:
I would go with the `_updateStageStatus` or `_setStageStatus` if you change the `enable` key to `status` which I think is a better option, then you avoid werid scenarios like `enable` is `disabled` ...